tests/test_simulation_engines.py

```diff
@@ -0,0 +1,36 @@
+"""
+Tests for simulation engine infrastructure.
+"""
+import unittest
+
+import stdpopsim
+
+
+class TestSimulationEngine_API(unittest.TestCase):
+    """
+    Tests for the API exposed for simulation engines.
+    """
+    def _test_simulation_engine(self, engine):
+        self.assertIsInstance(engine, stdpopsim.SimulationEngine)
+        self.assertIsNotNone(engine.simulate)
+        self.assertNotEqual(len(engine.citations), 0)
+
+    def test_get_default_simulation_engine(self):
+        engine = stdpopsim.get_default_simulation_engine()
+        self._test_simulation_engine(engine)
+
+    def test_all_simulation_engines(self):
+        for engine in stdpopsim.all_simulation_engines():
+            self._test_simulation_engine(engine)
+
+    def test_register_engine(self):
```

Comment:
This tests has a side-effect, which might bite us later. Not sure what do about it to be honest; might be OK to just drop the test.

Reply:
This is just within the scope of the running tests, though, right? There is a (somewhat hackish) way to remove this, because the underscored dict is only private by convention.
```
import stdpopsim.simulation_engines
del stdpopsim.simulation_engines._registered_simulation_engines["test-engine"]
```

A test like this could be added for `register_species()` too.

Reply:
Yes, it's just in the context of running the tests. I guess we could clean it up like you say. Another thing we could do is maybe mock that variable or something?